### PR TITLE
B1.A Backend PUT operations on Course Controller for Courses, Course Staff

### LINKED ISSUE #6: Issue B1.A: Backend PUT operations, Course Controller for Courses, Course Staff
- [ ] There is a PUT endpoint /api/course/update?id=123 where you can update a course but only if you are either (a) an admin, or (b) an instructor AND your user id appears in the courseStaff table for the course.

### PR DESCRIPTION
**Pull request by dayvidwang**
This PR adds an update(PUT) endpoint for courses along with associated tests. This endpoint is restricted to users only if they have the ADMIN role or if they have the INSTRUCTOR role and they are listed in the course staff for the targeted course.

Closes #6

### CODE REVIEW

**Review by darylou — DISMISSED** [review]
LGTM!

File: src/main/java/edu/ucsb/cs156/organic/controllers/CoursesController.java [hunk 1]
```diff
@@ -132,4 +132,55 @@ public Iterable<Staff> getStaff(
...
+            if (staff.getGithubId().equals(u.getGithubId())) {
+                isStaffOrAdmin = true;
+                break;
+            }
+        }
+
+        if (!isStaffOrAdmin) {
+            throw new AccessDeniedException(
+                    "User is not a staff member for this course");
+        }
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/organic/controllers/CoursesController.java]: Please see: https://ucsb-cs156.github.io/topics/sql/sql_queries.html

That article was made exactly for this situation.

There's a better approach here that replaces 14 lines of code with if tests and for loops with five lines of straight line code.

**Review by pconrad — CHANGES_REQUESTED** [review]
Looks good for the most part, but there is one place where the code can be greatly simplified by using the power of the Spring framework...

**Review by darylou — APPROVED** [review]
LGTM!

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**